features/request_specs/request_spec.feature

```diff
@@ -17,8 +17,10 @@ Feature: Request specs
 
   RSpec provides two matchers that delegate to Rails assertions:
 
-      render_template # delegates to assert_template
-      redirect_to     # delegates to assert_redirected_to
+```ruby
+render_template # delegates to assert_template
+redirect_to     # delegates to assert_redirected_to
+```
```

Comment:
Features should rename indented with their surrounding text, `e.g. RSpec` here, they're not markdown and expect indentation.

Reply:
Good evening, thanks for the feedback!

Originally I had these indented at the same level as the surrounding text, however that led to extra spaces at the beginning of the line that @pirj noted. I believe this stems from two aspects of gherkin (and I am not trying to second guess your knowledge of features or Gherkin or anything like that. You all have much greater command of it than I and I'm happy to hear where I'm wrong), which ultimately led to me rendering them as markdown and outdented as far as I did:

1. The [Feature section](https://cucumber.io/docs/gherkin/reference/#feature) of Gherkin assumes what is in its description is free-form text until it hits another section heading (such as Scenario in this case). I believe this means that DocStrings aren't recognized as such in that section, so the syntax that is working for Doc Strings later in the Feature won't work up in the description portion.
3. [Doc Strings](https://cucumber.io/docs/gherkin/reference/#examples) (scroll down one paragraph from the link) can be either three double-quotes or three backticks. Indentation of the opening quotes or backticks is explicitly unimportant, what matters is indentation of what is inside the quotes. However, due to the caveat above they aren't being recognized in the Feature description portion, so they have to be recognized as Markdown code blocks, if there will be a code block in a Feature's description.

Given the above, I used typical Markdown fenced codeblock syntax, not a Gherkin DocString and outdented the code all the way to the left so that there wouldn't be any extra spaces at the beginning. I could reindent them all one block in and have two extra spaces at the front of each line in the code block and a different solution altogether would be to revert those boxes off of being code boxes and turn them into just block comments. The only reason I went down this path of making them functioning code blocks is that they currently are broken code blocks, so I figured that was the desired path (for example [here](https://rspec.info/features/3-12/rspec-expectations/built-in-matchers/predicates/)). Please let me know which of those three solutions you would prefer! And if there's another, better one I haven't considered, please let me know that as well!

The workflow I am using to test these changes is the rspec-dev pipeline (make edits within the `repos/rspec-rails` folder, run `rake update_docs`, have rspec.github.io running in LIVERELOAD mode). My hope is it will be close to what you will see when you publish documentation manually. 

In closing, the main reason I care to do this is I have been enjoying ya'lls package as I add a bunch of tests to one of my packages, thank you for writing RSpec! I would love to be looking at syntax highlighting in every code example I read as I go through the documentation. I figured if I would take the work to make syntax highlighting work, I might as well contribute it back. I certainly do not want to put the documentation in any state you're not looking for it to be in and I easily could have some of the above mistaken. I would love to learn the right answer!

Have a great evening!

Reply:
@pirj noted the indentation was wrong in `md` files, not `feature` files, only feature files need to be changed back to being indented inline with their body.

Reply:
Ok! I've pushed the corresponding fix for the feature files. Have a great evening!